izpack-installer/src/main/java/com/izforge/izpack/installer/container/provider/GUIInstallDataProvider.java

```diff
@@ -375,16 +370,13 @@ private void loadGUIInstallData(GUIInstallData installData, Resources resources)
     {
         installData.guiPrefs = (GUIPrefs) resources.getObject("GUIPrefs");
     }
+    private static int getJavaMajorVersion() {
+        String v = System.getProperty("java.version");
 
-    private static int getJavaMajorVersion(String versionString) {
-        if (versionString == null || versionString.isEmpty())
-            return 1;
-        if (versionString.startsWith("1.")) {
-            return versionString.charAt(2) - '0';
-        } else {
-            String[] split = versionString.split("[.+_-]", 2);
-            return Integer.parseInt(split[0]);
+        if (v.startsWith("1.")) {
+            v = v.substring(2);
         }
+        String[] split = v.split("[.+_-]", 2);
+        return Integer.parseInt(split[0]);
     }
```

Comment:
You could call the method `getCurrentMajorVersion()` from `izpack-tools/src/main/java/com/izforge/izpack/util/JavaVersion.java`  outlined above...

Reply:
Done. I mimic API of `java.lang.Runtime.Version`